lnbits/wallets/corelightning.py

```diff
@@ -111,94 +122,111 @@ async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse
+            return PaymentResponse(False, None, None, None, f"Payment failed: '{exc}'.")
 
     async def get_invoice_status(self, checking_id: str) -> PaymentStatus:
         try:
             r: dict = self.ln.listinvoices(payment_hash=checking_id)  # type: ignore
-        except RpcError:
-            return PaymentPendingStatus()
-        if not r["invoices"]:
-            return PaymentPendingStatus()
 
-        invoice_resp = r["invoices"][-1]
-
-        if invoice_resp["payment_hash"] == checking_id:
-            if invoice_resp["status"] == "paid":
-                return PaymentSuccessStatus()
-            elif invoice_resp["status"] == "unpaid":
+            if not r["invoices"]:
                 return PaymentPendingStatus()
-            elif invoice_resp["status"] == "expired":
-                return PaymentFailedStatus()
-        else:
-            logger.warning(f"supplied an invalid checking_id: {checking_id}")
-        return PaymentPendingStatus()
 
-    async def get_payment_status(self, checking_id: str) -> PaymentStatus:
-        try:
-            r: dict = self.ln.listpays(payment_hash=checking_id)  # type: ignore
-        except Exception:
+            invoice_resp = r["invoices"][-1]
+
+            if invoice_resp["payment_hash"] == checking_id:
+                if invoice_resp["status"] == "paid":
+                    return PaymentSuccessStatus()
+                elif invoice_resp["status"] == "unpaid":
+                    return PaymentPendingStatus()
+                elif invoice_resp["status"] == "expired":
+                    return PaymentFailedStatus()
+            else:
+                logger.warning(f"supplied an invalid checking_id: {checking_id}")
             return PaymentPendingStatus()
-        if "pays" not in r:
+        except RpcError as exc:
+            logger.warning(exc)
+            return PaymentPendingStatus()
+        except Exception as exc:
+            logger.warning(exc)
             return PaymentPendingStatus()
-        if not r["pays"]:
-            # no payment with this payment_hash is found
-            return PaymentFailedStatus()
-
-        payment_resp = r["pays"][-1]
 
-        if payment_resp["payment_hash"] == checking_id:
-            status = payment_resp["status"]
-            if status == "complete":
-                fee_msat = -int(
-                    payment_resp["amount_sent_msat"] - payment_resp["amount_msat"]
-                )
+    async def get_payment_status(self, checking_id: str) -> PaymentStatus:
+        try:
+            r: dict = self.ln.listpays(payment_hash=checking_id)  # type: ignore
 
-                return PaymentSuccessStatus(
-                    fee_msat=fee_msat, preimage=payment_resp["preimage"]
-                )
-            elif status == "failed":
+            if "pays" not in r:
+                return PaymentPendingStatus()
+            if not r["pays"]:
+                # no payment with this payment_hash is found
```

Comment:
i fail like this is scary, what if you immediately check the status and the payment is not yet returned here.

Reply:
Nothing new here, the code has just been moved into a `try-catch` block, no functional changes made.